### ISSUE TITLE
Liquibase auto configuration not working with Liquibase 3.5.3 and YAML format

### ISSUE THREAD

**Issue by jramnani** [spring-projects/spring-boot#9971]
## Problem

Today I tried to get Spring Boot working with Liquibase database migrations as described in the [Spring Boot Database Initialization] documentation.  The documented steps didn't work for me with the following configuration:

* Spring Boot 1.5.4
* Liquibase 3.5.3

Here are the steps I followed:

* Added a dependency on `liquibase-core` to my Gradle build.
* Created a Liquibase database changelog file at `src/main/resources/db/changelog/db.changelog-master.yaml` (aka `classpath:db/changelog/db.changelog-master.yaml`)
* Start the application using the Gradle task, `bootRun`.

In the output of `bootRun` I noticed the following message:

```
:bootRunIgnoring Class-Path entry lib/snakeyaml-1.13.jar found in/Users/example/.gradle/caches/modules-2/files-2.1/org.liquibase/liquibase-core/3.5.3/f346404c3876c5bd0e07a7e10b7565bdcd35dbab/liquibase-core-3.5.3.jar as /Users/example/.gradle/caches/modules-2/files-2.1/org.liquibase/liquibase-core/3.5.3/f346404c3876c5bd0e07a7e10b7565bdcd35dbab/lib/snakeyaml-1.13.jar does not exist
```

With the root logging level set to DEBUG I can see Liquibase starting up, acquiring a database lock, then releasing the lock.  The changelog described in the YAML file is never run.

I believe this issue is caused by a bug in the packaging of Liquibase version 3.5.3, and is reported on their bug tracker: https://liquibase.jira.com/browse/CORE-3046.  The way this version of Liquibase is packaged, the YAML parsing library cannot be found.  I tried excluding that transitive dependency, and including an explicit dependency on the `snakeyaml` library, but that did not fix this issue for me.

## Workaround

I was able to get Liquibase migrations to work by doing the following:

Edit the `application.properties` file to explicitly specify the location of the Liquibase changelog in XML format.

```
liquibase.change-log=classpath:db/changelog/db.changelog-master.xml
```

I'm reporting this so that others may find this issue and the workaround.  Please let me know if I've made a mistake, and if there's somewhere else where this should be reported.

[Spring Boot Database Initialization]: https://docs.spring.io/spring-boot/docs/current/reference/html/howto-database-initialization.html#howto-execute-liquibase-database-migrations-on-startup

**Comment by wilkinsona** [comment]
That message can be benign. It's DevTools indicating that there's a problem with Liquibase's manifest as it references the SnakeYAML jar in a location where it doesn't exist. As long as SnakeYAML is on the class path via some other means things should still work. One way to get SnakeYAML on the class path is to depend on `spring-boot-starter` (either directly or transitively). That's what our [Liquibase sample](https://github.com/spring-projects/spring-boot/tree/v1.5.6.RELEASE/spring-boot-samples/spring-boot-sample-liquibase) does and it works fine with a YAML change log.

In short, the workaround that you have proposed should not be necessary with the required jars on the class path. I can't tell why you, apparently, do not have the required jars on the class path with the information you've provided. If you'd like us to investigate further, please provide a small sample that reproduces the behaviour you've described.

**Comment by spring-issuemaster** [comment]
If you would like us to look at this issue, please provide the requested information. If the information is not provided within the next 7 days this issue will be closed.

**Comment by jramnani** [comment]
@wilkinsona Thank you for responding to my issue.  I tried to duplicate this issue with a fresh checkout of my project, and I wasn't able to.

I ported my XML changelog over to YAML, started the server with `gradle bootRun`, and it worked.  The YAML changelog was discovered during application startup and the migrations were applied.

It sounds like we can close this issue, since I'm unable to reproduce.  While a little embarrassing, it looks like no harm done.  Thanks again for your help.

**Comment by wilkinsona** [comment]
@jramnani Thanks for following up. Glad to hear things are working for you now.
